onnxruntime/contrib_ops/cpu/moe/moe_quantization_cpu.cc

```diff
@@ -617,7 +651,15 @@ Status QMoECPU<T>::Compute(OpKernelContext* context) const {
           const int64_t end_idx = std::min(start_idx + dynamic_block_size, num_expert_tokens);
 
           for (int64_t i = start_idx; i < end_idx; ++i) {
-            const int64_t token_idx = routes[static_cast<size_t>(i)] / k_;
+            const int64_t route_idx = routes[static_cast<size_t>(i)];
+            // Validate route index to prevent division by zero or invalid token indices
+            if (route_idx < 0 || k_ <= 0) {
```

Comment:
Do we really need this check in the loop? For example, k_ <=0 should be check when we read k attribute. Not here.